engines/osxnative/sandbox.go

```diff
@@ -0,0 +1,147 @@
+package osxnative
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+// white list of environment variables that must
+// be passed to the child process
+var environmentWhitelist = []string{
+	"PATH",
+	"HOME",
+	"USER",
+	"SHELL",
+	"TMPDIR",
+	"PWD",
+	"EDITOR",
+	"LANG",
+	"LOGNAME",
+	"TERM",
+	"TERM_PROGRAM",
+}
+
+type sandbox struct {
+	context     *runtime.TaskContext
+	taskPayload *payload
+	env         *[]string
+	aborted     bool
+}
+
+func newSandbox(context *runtime.TaskContext, taskPayload *payload, env *[]string) *sandbox {
+	return &sandbox{
+		context,
+		taskPayload,
+		env,
+		false,
+	}
+}
+
+func downloadLink(link string) (string, error) {
```

Comment:
Couldn't we just use "curl" as one of the commands to get the script if we agree that at some point this is probably going away anyways?